app/models/user.rb

```diff
@@ -206,7 +207,42 @@ def self.authenticator(username = nil)
   end
 
   def self.authenticate(username, password, request = nil, options = {})
-    authenticator(username).authenticate(username, password, request, options)
+    user = authenticator(username).authenticate(username, password, request, options)
+    user.try(:link_user_to_session, request)
+
+    user
+  end
+
+  def link_user_to_session(request)
+    return unless request
+
+    session_id = request.session_options[:id]
+    sessions << Session.find_or_create_by(:session_id => session_id)
```

Comment:
So I am wondering if we should only do this for SQL sessions, and not for the other ones.  This works, and probably isn't a big deal since we are already making some SQL calls for this section, but it is using a different "store" when using `memcache` or `memory` currently.

Reply:
We did the creation of the session record as a way to record the list for later querying by user name.  Ideally, if each store had it's own way to search by user name, then this wouldn't be needed.  However, since neither memcache nor memory has this, we'd have to code it ourselves (and probably would have to do it for other ones like Redis too).  Instead we went for the "simple" route which is to just record it all in the Sessions table, regardless.

However, we do concede this may have performance impacts and may not be desirable.  Additionally, this particular line probably has to use a much smarter locking mechanism as find_or_create_by isn't exactly process/thread safe.

Reply:
Going to address this at a later date if we do decided to optimize this.  While not great, there is no real harm in doing this differently in the future and leaving some invalid `Session` records in the DB.